### PR TITLE
refactor: system tables in FrontendCatalogManager

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

- Removes some deprecated structs around `TableEngine`
- Add `SystemCatalog` (reuse the name from previous system catalog, but they are two different things) for accessing system tables
- Simplify trait `CatalogManager`. All register/deregister methods are removed as they are/will not go through the catalog manager

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Review by MichaelScofield — APPROVED** [review]
DLJB

**Review by fengjiachun — APPROVED** [review]
LGTM

File: src/frontend/src/catalog.rs [hunk 1]
```diff
@@ -334,3 +268,57 @@ impl CatalogManager for FrontendCatalogManager {
...
+    catalog_manager: Weak<FrontendCatalogManager>,
+}
+
+impl SystemCatalog {
+    fn schema_names(&self) -> Vec<String> {
+        vec![INFORMATION_SCHEMA_NAME.to_string()]
+    }
+
+    fn table_names(&self, schema: &str) -> Vec<String> {
+        if schema == INFORMATION_SCHEMA_NAME {
```
Comment (killme2008) [inline src/frontend/src/catalog.rs:290]: Why not use match? It's more clear and elegant.

File: src/frontend/src/catalog.rs [hunk 2]
```diff
@@ -334,3 +268,57 @@ impl CatalogManager for FrontendCatalogManager {
...
+            vec![]
+        }
+    }
+
+    fn schema_exist(&self, schema: &str) -> bool {
+        schema == INFORMATION_SCHEMA_NAME
+    }
+
+    fn table_exist(&self, schema: &str, table: &str) -> bool {
+        if schema == INFORMATION_SCHEMA_NAME {
```
Comment (killme2008) [inline src/frontend/src/catalog.rs:304]: what about `match (schema, table)` ?

File: src/catalog/src/local/memory.rs [hunk 3]
```diff
@@ -264,7 +218,7 @@ impl MemoryCatalogManager {
     }
 
     /// Registers a schema and returns an error if the catalog or schema does not exist.
-    pub fn register_table(&self, request: RegisterTableRequest) -> Result<bool> {
+    pub fn register_table_sync(&self, request: RegisterTableRequest) -> Result<bool> {
```
Comment (killme2008) [inline src/catalog/src/local/memory.rs:221]: The `sync` postfix is unnecessary. And `sync` has a different meaning in Rust that may cause confusion.

**Pull request merged**